modules/cache/jvm/src/main/scala/coursier/cache/ConnectionBuilder.scala

```diff
@@ -16,6 +16,7 @@ import dataclass.data
   autoCredentials: Seq[DirectCredentials] = Nil,
   sslSocketFactoryOpt: Option[SSLSocketFactory] = None,
   hostnameVerifierOpt: Option[HostnameVerifier] = None,
+  authRealmOpt: Option[String] = None,
```

Comment:
This field should be added in last position, after a `@since("2.0.17")` annotation, not to break binary compatibility.